### PR TITLE
WIP: 1-namespacing: renaming components and namespacing css

### PR DESCRIPTION
**Pull request by christopherdavidbach**
…t/.css

### CODE REVIEW

File: components/SitePost/index.jsx [hunk 1]
```diff
@@ -23,11 +23,11 @@ class SitePost extends React.Component {
         return (
             <div>
               { home }
-              <div className='blog-single'>
-                <div className='text'>
-                  <h1>{ post.title }</h1>
+              <div className='blog-entry'>
```
Comment (rcaracaus) [inline components/SitePost/index.jsx]: We should use the name of the component.

File: README.md [hunk 2]
```diff
@@ -5,7 +5,7 @@ Gatsby starter for creating a blog
 ## Features
 + Inline CSS.
 + Clean and extensible code.
-+ Six simple components: SiteSidebar, SiteNav, SiteLinks, SitePost, SitePage, ReadNext.
++ Six simple components: Sidebar, Nav, SiteLinks, Blog, Page, BlogDetail.
```
Comment (rcaracaus) [inline README.md]: We don't need to mess with this. We can delete this

File: components/BlogDetail/index.jsx [hunk 3]
```diff
@@ -5,34 +5,34 @@ import DocumentTitle from 'react-document-title'
 import { prefixLink } from 'gatsby-helpers'
 import access from 'safe-access'
 import { config } from 'config'
-import ReadNext from '../ReadNext'
-import './style.css'
+import BlogFooter from '../BlogFooter'
```
Comment (rcaracaus) [inline components/BlogDetail/index.jsx]: why did you change this from ReadNext?

Comment (rcaracaus) [inline components/BlogDetail/index.jsx]: Is there even a "Read Next" button? I can't see it in original Lumen? http://wpioneer.github.io/gatsby-starter-lumen/hello-world/ Link?

Comment (rcaracaus) [inline components/BlogDetail/index.jsx]: Oh I see, you just removed Read Next? Did it exist?

File: components/BlogDetail/index.jsx [hunk 4]
```diff
@@ -5,34 +5,34 @@ import DocumentTitle from 'react-document-title'
...
+import './style.scss'
 import '../../static/css/highlight.css'
 
-class SitePost extends React.Component {
+class BlogDetail extends React.Component {
```
Comment (rcaracaus) [inline components/BlogDetail/index.jsx]: You didn't really need to change the name of the component, you could have just changed the name of the CSS

File: components/BlogDetail/style.scss [hunk 5]
```diff
@@ -0,0 +1,126 @@
+.blog-teaser {
```
Comment (rcaracaus) [inline components/BlogDetail/style.scss]: The name of css namespace is "blog-teaser", it should match name of component

File: components/BlogDetail/index.jsx (hunk 3 above)
Comment (edisch) [inline components/BlogDetail/index.jsx]: It wasn't doing a "ReadNext" function it's just a footer to the blog with a plug for the author's twitter.

File: components/BlogDetail/index.jsx (hunk 4 above)
Comment (edisch) [inline components/BlogDetail/index.jsx]: I think it's probably better to keep our css and components named the same though for clarity. Would be weird to have blog-detail css referring to SitePost html elements. Just my opinion though.

File: components/BlogDetail/style.scss (hunk 5 above)
Comment (edisch) [inline components/BlogDetail/style.scss]: BlogTeaser actually should be a seperate component as it is just the little teaser on the front page. Unfortunately the html for BlogTeaser is currently getting spat out from the main index.jsx.

File: components/BlogDetail/style.scss [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+.blog-teaser__readmore {
+    font-size: 16px;
+    color: #5d93ff;
+}
+
+//
+
+.blog-detail__description {
+	font-size: responsive 12px 14px;
```
Comment (rcaracaus) [inline components/BlogDetail/style.scss]: indentation

File: components/BlogDetail/style.scss [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+    text-align: center;
+    padding: 8px 16px;
+    color: #999;
+    background: #f4f4f4;
+    font-weight: 400;
+    border-radius: 3px;
+
+    &:focus {
+		  color: #444;
+		  background: #f4f4f4;
```
Comment (rcaracaus) [inline components/BlogDetail/style.scss]: indentation

Comment (edisch) [inline components/BlogDetail/style.scss]: I think phpstorm and atom are conflicting. Some of the indents from chris' original commit came out weird on my end. This block here looks normal on my editor.

File: components/BlogDetail/style.scss (hunk 5 above)
Comment (rcaracaus) [inline components/BlogDetail/style.scss]: The css namespace should be the same name as the component it styles

File: components/BlogDetail/style.scss (hunk 7 above)
Comment (rcaracaus) [inline components/BlogDetail/style.scss]: Don't know what to say, needs to be fixed in Github

File: components/BlogDetail/style.scss [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+    text-align: center;
+    margin-top: 30px;
+}
+
+.blog-detail__text {
+    line-height: 31px;
+    margin-bottom: 31px;
+}
+
+.blog-detail__title, .blog-detail .blog-detail__date-published, .blog-detail .blog-detail__text *:not(div) {
```
Comment (rcaracaus) [inline components/BlogDetail/style.scss]: this should be using extends

File: components/BlogDetail/style.scss [hunk 9]
```diff
@@ -0,0 +1,82 @@
+.blog-detail__description {
```
Comment (rcaracaus) [inline components/BlogDetail/style.scss]: why is indentation still messed up?

File: components/BlogDetail/style.scss [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+    margin-bottom: 31px;
+}
+
+.blog-detail__title, .blog-detail .blog-detail__date-published, .blog-detail .blog-detail__text *:not(div) {
+    max-width: 630px;
+    margin-left: auto;
+    margin-right: auto;
+}
+
+.blog-detail .blog-detail__text img {
```
Comment (rcaracaus) [inline components/BlogDetail/style.scss]: we should not need to use tags if we have full access to html templates

File: components/BlogDetail/style.scss [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+		}
+
+		&:hover {
+			color: #444;
+			background-color: #f4f4f4;
+			border: 0;
+		}
+}
+
+@media (max-width:900px) {
```
Comment (rcaracaus) [inline components/BlogDetail/style.scss]: this should be using @breakpoint inside the respective components

File: components/BlogTeaser/index.jsx [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+    const description = access(page, 'data.description')
+    const datePublished = access(page, 'data.date')
+    const category = access(page, 'data.category')
+
+    return (
+      <div className='blog-teaser'>
+        <div className="blog-teaser__time"time dateTime={ moment(datePublished).format('MMMM D, YYYY') }>
+          { moment(datePublished).format('MMMM YYYY') }
+        </div>
+        <span style={ {padding: '5px'} }></span>
```
Comment (rcaracaus) [inline components/BlogTeaser/index.jsx]: what is up with wack inline styles?

File: components/BlogDetail/style.scss (hunk 10 above)
Comment (edisch) [inline components/BlogDetail/style.scss]: leftovers from lumen we haven't namespaced the img's yet

**Comment by rcaracaus** [comment]
I would make sure components PR is 100% before I looked at scss lint or you are going to have conflict hell

File: components/Sidebar/style.css [hunk 13]
```diff
@@ -1,11 +1,11 @@
 .sidebar {
 	lost-column: 1/3;
 }
-.sidebar .sidebar-inner {
+.sidebar .sidebar__inner {
 	position: relative;
 	padding: 40px;
 }
-.sidebar .sidebar-inner:after {
+.sidebar .sidebar__inner:after {
```
Comment (rcaracaus) [inline components/Sidebar/style.css]: why .sidebar .sidebar__inner      redundant

File: pages/index.jsx [hunk 14]
```diff
@@ -18,32 +18,17 @@ class SiteIndex extends React.Component {
...
             }
         })
 
         return (
             <DocumentTitle title={ config.siteTitle }>
               <div>
-                <SiteHeader />
-                <SiteSidebar {...this.props}/>
+                <Header/>
+                <Sidebar {...this.props}/>
```
Comment (rcaracaus) [inline pages/index.jsx]: can you explain the props shit to me tomorrow

File: README.md [hunk 15]
```diff
@@ -5,7 +5,7 @@ Gatsby starter for creating a blog
 ## Features
 + Inline CSS.
 + Clean and extensible code.
-+ Six simple components: SiteSidebar, SiteNav, SiteLinks, SitePost, SitePage, ReadNext.
++ Six simple components: Sidebar, Nav, SiteLinks, BlogFooter, Page, BlogDetail.
```
Comment (rcaracaus) [inline README.md]: you don't ned to update this, we should probably remove this line and other things we don't care about

File: components/BlogDetail/style.scss (hunk 9 above)
Comment (rcaracaus) [inline components/BlogDetail/style.scss]: indentation still not fixed

File: components/BlogDetail/style.scss (hunk 8 above)
Comment (rcaracaus) [inline components/BlogDetail/style.scss]: still not using extends

File: components/BlogDetail/style.scss [hunk 16]
```diff
@@ -1,12 +1,12 @@
...
+.BlogDetail__description {
   font-size: responsive 12px 14px;
   line-height: 1.6;
 }
 
-.blog-detail {
+.BlogDetail {
   @extend %LayoutMax;
   max-width: 945px;
   margin-left: auto;
```
Comment (rcaracaus) [inline components/BlogDetail/style.scss]: this is defined in LayoutMax

File: components/BlogDetail/style.scss [hunk 17]
```diff
@@ -42,11 +42,11 @@
...
 }
 
-.blog-detail__date-published {
+.BlogDetail__date-published {
   @extend %LayoutMax;
 }
 
-.blog-detail__footer {
+.BlogDetail__footer {
   max-width: 630px;
```
Comment (rcaracaus) [inline components/BlogDetail/style.scss]: this can just extend LayoutMax

File: components/BlogTeaser/style.scss [hunk 18]
```diff
@@ -1,34 +1,34 @@
...
+.BlogTeaser__title {
     font-family: Amatica SC, sans-serif;
     font-size: 1.6875rem;
     margin-top: 10px;
     line-height: 30px;
     margin-bottom: 20px;
 }
 
-.blog-teaser__tease {
+.BlogTeaser__tease {
```
Comment (rcaracaus) [inline components/BlogTeaser/style.scss]: teaser**

File: components/Nav/style.scss [hunk 19]
```diff
@@ -1,24 +1,30 @@
...
 	font-size: 16px;
-	line-heigh: 26px;
+	line-height: 26px;
 	margin-bottom: 26px;
 	border-bottom: 0;
 	font-weight: 400;
 	color: #222;
 }
-.nav ul li a.current {
+	a:hover {
```
Comment (rcaracaus) [inline components/Nav/style.scss]: couldn't we just put the class on the link directly?

File: components/Sidebar/style.scss [hunk 20]
```diff
@@ -0,0 +1,73 @@
...
+	}
+
+	@include breakpoint(1100px) {
+		padding: 35px 20px 0;
+		lost-column: 1/2;
+	}
+}
+
+.Sidebar__inner {
+	position: relative;
```
Comment (rcaracaus) [inline components/Sidebar/style.scss]: can we use two spaces for indentation instead of 4? Lots of space

File: components/Nav/style.scss (hunk 19 above)
Comment (edisch) [inline components/Nav/style.scss]: yea it should be &:hover inside the class

File: components/Sidebar/style.scss [hunk 21]
```diff
@@ -0,0 +1,73 @@
...
+		 position: absolute;
+		 content: '';
+		 width: 1px;
+		 height: 540px;
+		 top: 30px;
+		 right: -10px;
+		 bottom: 0;
+	}
+
+	@include breakpoint(500px) {
```
Comment (rcaracaus) [inline components/Sidebar/style.scss]: do we need all these breakpoints? A little rediculous

File: components/Social/style.css [hunk 22]
```diff
@@ -16,17 +16,17 @@
 	border-radius: 3px;
 	background: #f4f4f4;
 }
-.social ul > li:hover {
+.Social ul > li:hover {
 	background: #f4f4f4;
 }
-.social ul > li > a {
+.Social ul > li > a {
```
Comment (rcaracaus) [inline components/Social/style.css]: why do we need to use tags here?

**Comment by rcaracaus** [comment]
Sidebar is messed up

screen shot 2016-06-30 at 4 21 18 pm

File: README.md (hunk 15 above)
Comment (rcaracaus) [inline README.md]: We can remove anythng that is gatsby specific

File: components/Page__BlogListing/index.jsx [hunk 23]
```diff
@@ -0,0 +1,49 @@
+import React from 'react'
```
Comment (rcaracaus) [inline components/Page__BlogListing/index.jsx]: What is Page__BlogListing? Why such a crzy name?

File: components/Sidebar/style.scss [hunk 24]
```diff
@@ -0,0 +1,49 @@
...
+
+.Sidebar {
+	display: block;
+	position: relative;
+	float: none;
+	padding: 1rem;
+	margin: 0 1rem;
+	border-bottom: 1px solid #ccc;
+
+	@include breakpoint(800px) {
```
Comment (rcaracaus) [inline components/Sidebar/style.scss]: this is all crazy but will be fixed by scss linter next PR

File: components/Social/index.jsx [hunk 25]
```diff
@@ -0,0 +1,40 @@
...
+    render() {
+
+        return (
+            <div className='Social'>
+              <ul className="Social__list">
+                <li className="Social__list-item">
+                  <a className="Social__item-link" href={ config.siteTwitterUrl }><i className='fa fa-twitter Social__icon'></i></a>
+                </li>
+                <li className="Social__list-item">
+                  <a className="Social__item-link" href={ config.siteGithubUrl }><i className='fa fa-github-alt Social__icon'></i></a>
```
Comment (rcaracaus) [inline components/Social/index.jsx]: Just -link  not item-link

File: components/Page__BlogListing/index.jsx (hunk 23 above)
Comment (edisch) [inline components/Page__BlogListing/index.jsx]: Homepage has special functionality by listing blogs so it's a Page of BlogListing hence Page__BlogListing

File: components/Social/index.jsx [hunk 26]
```diff
@@ -0,0 +1,40 @@
...
+import './style.scss'
+import '../../static/fonts/fontawesome/style.css'
+
+class Social extends React.Component {
+    render() {
+
+        return (
+            <div className='Social'>
+              <ul className="Social__list">
+                <li className="Social__list-item">
```
Comment (rcaracaus) [inline components/Social/index.jsx]: why list-item? Why not just "item"?
